### PR TITLE
Implement the OutputPath variable

### PR DESCRIPTION
**Pull request by SamErde**
Implements the OutputPath variable that was previously defined. Allows the op to specify a path to store the Locksmith CSV files in. (Defaults to the current working directory.) Also prefixes the CSVs with "Locksmith" and the current timestamp.

### CODE REVIEW

File: Public/Invoke-Locksmith.ps1 [hunk 1]
```diff
@@ -109,6 +118,9 @@
     }
 
     ### Initial variables
+    # For output filenames
+    [string]$FilePrefix = "Locksmith $(Get-Date -format 'yyyy-MM-dd hh-mm-ss') "
```
Comment (TrimarcJake) [inline Public/Invoke-Locksmith.ps1:122]: Trailing space here creates a double-space in the output file names.

File: Public/Invoke-Locksmith.ps1 [hunk 2]
```diff
@@ -72,7 +72,16 @@
         [Parameter()]
             [ValidateSet('Auditing','ESC1','ESC2','ESC3','ESC4','ESC5','ESC6','ESC8','All','PromptMe')]
             [array]$Scans = 'All',
-        [string]$OutputPath = (Get-Location).Path,
+        
+        # The directory to save the output in (defaults to the current working directory).
+        [Parameter()]
+            [ValidateScript({(Test-Path -Path $_ -PathType Container)},
+                ErrorMessage = "{0} is not a valid directory path."
```
Comment (TrimarcJake) [inline Public/Invoke-Locksmith.ps1:79]: I don't think the ErrorMessage property is supported in Windows PowerShell 5.1. Works well in 7.4 tho

**Review by TrimarcJake — CHANGES_REQUESTED** [review]
Almost there. Need to improve PS5.1 support and clean up the filename prefix.

File: Public/Invoke-Locksmith.ps1 (hunk 2 above)
Comment (SamErde) [reply Public/Invoke-Locksmith.ps1:79]: Ah, you're right. Will remove.

**Comment by SamErde** [comment]
Changes made. Ready for review again.

**Review by TrimarcJake — APPROVED** [review]
YES. Tysm!

**Pull request merged**